receiver/postgresqlreceiver/scraper.go

```diff
@@ -0,0 +1,344 @@
+	dataPoint := metric.AppendEmpty()
+	dataPoint.SetTimestamp(ts)
+	dataPoint.SetIntVal(value)
+	if attributes.Len() > 0 {
+		attributes.CopyTo(dataPoint.Attributes())
+	}
+}
+
+// scrape scrapes the metric stats, transforms them and attributes them into a metric slices.
+func (p *postgreSQLScraper) scrape(ctx context.Context) (pdata.Metrics, error) {
+	// metric initialization
+	md := pdata.NewMetrics()
+	ilm := md.ResourceMetrics().AppendEmpty().InstrumentationLibraryMetrics().AppendEmpty()
+	ilm.InstrumentationLibrary().SetName("otel/postgresql")
+	now := pdata.NewTimestampFromTime(time.Now())
+
+	blocksRead := initMetric(ilm.Metrics(), metadata.M.PostgresqlBlocksRead).Sum().DataPoints()
+	commits := initMetric(ilm.Metrics(), metadata.M.PostgresqlCommits).Sum().DataPoints()
+	databaseSize := initMetric(ilm.Metrics(), metadata.M.PostgresqlDbSize).Gauge().DataPoints()
+	backends := initMetric(ilm.Metrics(), metadata.M.PostgresqlBackends).Gauge().DataPoints()
+	databaseRows := initMetric(ilm.Metrics(), metadata.M.PostgresqlRows).Gauge().DataPoints()
+	operations := initMetric(ilm.Metrics(), metadata.M.PostgresqlOperations).Sum().DataPoints()
+	rollbacks := initMetric(ilm.Metrics(), metadata.M.PostgresqlRollbacks).Sum().DataPoints()
+
+	var errors scrapererror.ScrapeErrors
+
+	databases := p.config.Databases
+	listClient, err := p.clientFactory.getClient(p.config, "")
+	if err != nil {
+		p.logger.Error("Failed to initialize connection to postgres", zap.Error(err))
+		errors.Add(err)
+		return md, errors.Combine()
+	}
+	defer listClient.Close()
+
+	if len(databases) == 0 {
+		dbList, err := listClient.listDatabases(ctx)
+		if err != nil {
+			p.logger.Error("Failed to request list of databases from postgres", zap.Error(err))
+			errors.Add(err)
+			return md, errors.Combine()
+		}
+
+		databases = dbList
+	}
+
+	p.collectCommitsAndRollbacks(ctx, now, listClient, databases, commits, rollbacks, errors)
+	p.collectDatabaseSize(ctx, now, listClient, databases, databaseSize, errors)
+	p.collectBackends(ctx, now, listClient, databases, backends, errors)
+
+	for _, database := range databases {
+		dbClient, err := p.clientFactory.getClient(p.config, database)
+		if err != nil {
+			errors.AddPartial(2, err)
+			p.logger.Error("Failed to initialize connection to postgres", zap.String("database", database), zap.Error(err))
+			continue
+		}
+		defer dbClient.Close()
+
+		p.collectBlockReads(ctx, now, dbClient, blocksRead, errors)
+		p.collectDatabaseTableMetrics(ctx, now, dbClient, databaseRows, operations, errors)
+	}
+
+	return md, errors.Combine()
+}
+
+func (p *postgreSQLScraper) collectBlockReads(
+	ctx context.Context,
+	now pdata.Timestamp,
+	client client,
+	blocksRead pdata.NumberDataPointSlice,
+	errors scrapererror.ScrapeErrors,
+) {
+	blocksReadByTableMetrics, err := client.getBlocksReadByTable(ctx)
+	if err != nil {
+		p.logger.Error("Errors encountered while fetching blocks read by table", zap.Error(err))
+		errors.AddPartial(1, err)
```

Comment:
I believe the first parameter here is supposed to represent a count of how many metrics failed to collect.